### ISSUE TITLE
Unify visual separator between mover buttons when show button label is on

### ISSUE THREAD

**Issue by t-hamano** [WordPress/gutenberg#58999]
### Description

As mentioned in #49556, there should be a visual separator between mover buttons when "show button label" is on. However, depending on the browser size, block orientation, and whether the top toolbar is enabled, this separator may not display correctly.

### Desktop view and horizontal layout

- Separator should be vertical.

![image](https://github.com/WordPress/gutenberg/assets/54422211/353202ea-4234-48ca-810b-d391b2b32669)

### Desktop view, horizontal layout, and top toolbar enabled

- Separator should be vertical.

![image](https://github.com/WordPress/gutenberg/assets/54422211/8950618f-53c8-4808-bac4-3edf3583b04b)

### Tablet view

- Separator color should be gray instead of black.

![image](https://github.com/WordPress/gutenberg/assets/54422211/bb886b56-366b-4bbc-83bb-a21fb404eefc)

### Tablet view and horizontal layout

- Separator should be vertical.
- Separator color should be gray instead of black.

![image](https://github.com/WordPress/gutenberg/assets/54422211/4f2c0b6c-bb51-46ce-82b9-66d18f314bfa)

### Mobile view

- There is no separator itself.

![image](https://github.com/WordPress/gutenberg/assets/54422211/86c6aa25-2fba-4d28-84dd-fa3410a489e8)

### Step-by-step reproduction instructions

- Open the post editor or site editor.
- Turn on "Show button text labels" from Options > Preferences > Accessibility.
- Select the block and check the toolbar.

### Environment info

- WordPress version: 6.5-beta1-57631
- Gutenberg version: 17.7.0-rc.1 

Regarding this issue, #49556 and #57640 are related, but not all issues were introduced in WP6.5. It may not be a necessary task for WP6.5, but if we can fix it, I think it's worth backporting.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by t-hamano** [comment]
@afercia 

Is a visual separator still necessary when mover buttons are horizontally aligned? I would appreciate any advice from an accessibility perspective 🙏

![mover-button](https://github.com/WordPress/gutenberg/assets/54422211/52e59d98-af51-4fe2-8bc6-ba921eb1d35a)

**Comment by afercia** [comment]
Ouch I missed these buttons change to 'Move left' and 'Move right' for blocks that can be rearranged horizontally 🤦🏽 

> Is a visual separator still necessary when mover buttons are horizontally aligned?

Generally, from an a11y perspective, I don't like buttons with no borders. Ideally, all buttons should have borders, which would make separators pointless.

I'd say that for now we should make them look consistently with other buttons _groups_ where separators are only used between groups while buttons don't have separators. For example in a Heading block:

![Screenshot 2024-02-14 at 12 19 37](https://github.com/WordPress/gutenberg/assets/1682452/e24e14c8-78e9-47a2-85c1-a8940c47f95f)

The format buttons don't have separators becouse they belong to the same group.
From a logical perspective, Move left / Move right belong to.a group so they should look consistently.

Aside: I don't understand why 'Change level' and  'Align text' are in the same group. Their purpose and functionality is different and they shouldn't be in the same group. I see they are in the same group also with icon buttons...

![Screenshot 2024-02-14 at 12 24 59](https://github.com/WordPress/gutenberg/assets/1682452/512d654d-caf1-4d7f-aeb9-4ae4965d564a)

**Comment by t-hamano** [comment]
Thank you for your reply.

In that case, do you think the following is the best solution at this point?

- If the mover button is _horizontal_, the separator is not always displayed.
- If the mover button is _vertical_, always show a separator so that the button is not misunderstood as having multiple lines of text ([#57640](https://github.com/WordPress/gutenberg/pull/57640) mentioned it)

**Comment by afercia** [comment]
Yes I think for now that's the best path forward 👍🏽
